### PR TITLE
feature/jl/monday_hardening

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. CRITICAL [LanguageParamUtils.java#L53](https://github.com/SonarSource/sonarqube/blob/f4060a1b8d71d77364e9e24bda35e1ea72d9be27/server/sonar-server/src/main/java/org/sonar/server/util/LanguageParamUtils.java#L53): Remove this "Nonnull" annotation to honor the overridden method's contract. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2638)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. CRITICAL [LanguageParamUtils.java#L53](https://github.com/SonarSource/sonarqube/blob/9c417604ad3c2877b3d69570cff829f96e3fccec/server/sonar-server/src/main/java/org/sonar/server/util/LanguageParamUtils.java#L53): Remove this "Nonnull" annotation to honor the overridden method's contract. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2638)

File: server/sonar-server/src/main/java/org/sonar/server/user/UserUpdater.java [hunk 1]
```diff
@@ -61,7 +61,7 @@
   private static final String NAME_PARAM = "Name";
   private static final String EMAIL_PARAM = "Email";
 
-  private static final int LOGIN_MIN_LENGTH = 3;
+  private static final int LOGIN_MIN_LENGTH = 2;
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/user/UserUpdater.java]: I'm surprised how short the fix was, that's a very good point!

File: server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/RestoreBuiltInActionTest.java [hunk 2]
```diff
@@ -19,40 +19,37 @@
...
 
-  @Before
-  public void setUp() {
-    tester = new WsTester(new QProfilesWs(
-      mock(RuleActivationActions.class),
-      mock(BulkRuleActivationActions.class),
-      mock(ProjectAssociationActions.class),
-      new RestoreBuiltInAction(profileService)));
+  @Test
+  public void return_empty_result_when_no_infos_or_warnings() {
```
Comment (teryk) [inline server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/RestoreBuiltInActionTest.java]: `when_no_info_or_warning` information is a plural

File: server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/RestoreBuiltInActionTest.java [hunk 3]
```diff
@@ -19,40 +19,37 @@
...
+    TestResponse response = tester.newRequest().setParam("language", "xoo").execute();
+    assertThat(response.getStatus()).isEqualTo(204);
```
Comment (teryk) [inline server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/RestoreBuiltInActionTest.java]: I would test the mock is called with the language

**Comment by teryk** [comment]
look at my feedback and LGTM

File: server/sonar-server/src/main/java/org/sonar/server/user/UserUpdater.java (hunk 1 above)
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/user/UserUpdater.java]: Indeed, good design -> easy maintainability !

**Pull request merged**